### DISCUSSION TITLE
Raspberry Pi - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#5240] [Docs]
# Raspberry Pi - Ultralytics YOLOv8 Docs

Quick start guide to setting up YOLO on a Raspberry Pi with a Pi Camera using the libcamera stack. Detailed comparison between Raspberry Pi 3, 4 and 5 models.

https://docs.ultralytics.com/guides/raspberry-pi/

**Comment by miguelsonohara** [comment]
I want to get ultralytics for my raspberry pi 4 but when I type "pip3 install ultralytics" to terminal, its just show up like this:
pip3 install ultralytics
error: externally-managed-environment

— This environment is externally managed
    To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

**Reply by SOTNER** [reply]
A temporary solution that helped me:
pip install pip==22.3.1 --break-system-packages

**Reply by pderrenger** [reply]
@SOTNER it looks like you've encountered an issue with the package management on your Raspberry Pi. The error message suggests that your environment is managed by the system package manager (`apt`), and it's generally not recommended to mix `pip` installations with system package installations to avoid conflicts.

The temporary solution you've found uses the `--break-system-packages` flag to override this protection, which can be risky and might lead to system instability or conflicts.

A safer approach would be to follow the recommended practice of creating a virtual environment. This isolates your Python environment and allows you to install packages without affecting the system-wide Python installation. Here's how you can do it:

1. Install `python3-venv` if it's not already installed:
   ```bash
   sudo apt update
   sudo apt install python3-venv python3-pip
   ```

2. Create a virtual environment:
   ```bash
   python3 -m venv my-yolov8-env
   ```

3. Activate the virtual environment:
   ```bash
   source my-yolov8-env/bin/activate
   ```

4. Now, within the virtual environment, you can safely use `pip` to install packages:
   ```bash
   pip install ultralytics
   ```

Remember to activate the virtual environment (`source my-yolov8-env/bin/activate`) every time you start a new terminal session and want to work with YOLOv8.

For more detailed instructions and best practices, you can refer to our documentation. 😊👍
